pkg/sql/logictest/testdata/logic_test/udf

```diff
@@ -3466,3 +3466,32 @@ SELECT f100915(20-(SELECT y FROM (VALUES (10), (20)) y(y) WHERE x=y)) FROM (VALU
 ----
 true
 true
+
+# Tests that document surprising but correct UDF behavior.
+subtest surprising_behavior
+
+statement ok
+CREATE TABLE t_param_names (a INT PRIMARY KEY, b INT);
+
+statement ok
+INSERT INTO t_param_names VALUES (1,2),(3,4);
+
+statement ok
+CREATE FUNCTION f_param_names(a INT, b INT) RETURNS INT AS
+$$
+  SELECT t_param_names.a FROM t_param_names WHERE t_param_names.a=a AND t_param_names.b=b;
+$$ LANGUAGE SQL;
+
+query I
+SELECT f_param_names(1,2);
+----
+1
+
+# When function params have the same name as the table columns the UDF
+# references, the results are counterintuitive. In this example, the UDF returns
+# a value from the first row in the primary key, even though the arguments
+# passed to it do not match. This matches postgres behavior.
```

Comment:
It'd be helpful to explain why this is happening. The `a` and `b` in the `WHERE` clause resolve to the table columns, not the function arguments. This is explain in the [Postgres docs](https://www.postgresql.org/docs/15/xfunc-sql.html):

> If the argument name is the same as any column name in the current SQL command within the function, the column name will take precedence. To override this, qualify the argument name with the name of the function itself, that is function_name.argument_name. (If this would conflict with a qualified column name, again the column name wins. You can avoid the ambiguity by choosing a different alias for the table within the SQL command.)

We haven't implemented the ability to do `function_name.argument_name` yet. I created #102802 to track this.